### PR TITLE
Simplify cachetasks

### PR DESCRIPTION
**Pull request by marcoscaceres**
- Drops message passing from SW and page :+1: 
- Simplifies SW code :+1: 
- Tests, as much as possible given current set up, cachetasks.js (~98% coverage).
- Removed proxies from Karma 
- Moved sw.js to js/sw.js + made all the necessary changes to accommodate that

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@oyiptong, @emtwo, @sarracini, r? I made this changes somewhat in a rush, so there is likely to be some mistakes/typos, etc. Please check... 

Also, yeah... I need to rebase.

**Comment by marcoscaceres** [comment]
Oki, rebased :dancer:

**Comment by marcoscaceres** [comment]
@oyiptong moved sw.js to js/, which gets rid of the Karma proxies also. **However**, please make sure the "Service-Worker-Allowed" is set appropriately on the CDN. Will add a bug.

**Comment by marcoscaceres** [comment]
I added a little Express server.js under the :file_folder: `bin`. You can then

``` BASH
npm run server
```

when developing.

**Comment by oyiptong** [comment]
will add the header

**Comment by marcoscaceres** [comment]
(splitting this up, as it's huge! Sorry everyone.... it's what happens when you put me on a plane without network... I got refactoring all the things)

**Comment by marcoscaceres** [comment]
I moved the little development server to a different pull request.

**Comment by marcoscaceres** [comment]
Ok, this is probably as much as I can squish the history without losing too much information. I want to keep some of the things that I added and later removed, as I have a feeling I might need them again later.

File: src/js/mainSiteURLs.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+[
```
Comment (marcoscaceres) [inline src/js/mainSiteURLs.json:1]: Making this JSON assures we only load it when the service worker is installed. Before, we were loading it every time the service worker became active.

**Comment by marcoscaceres** [comment]
I reverted moving the sw.js file. Made a "move_sw_to_src_dir" branch.

File: src/js/mainSiteURLs.json (hunk 1 above)
Comment (oyiptong) [inline src/js/mainSiteURLs.json:1]: :+1:

File: src/js/sites.js [hunk 2]
```diff
@@ -198,11 +199,9 @@
 
       // generate a regular thumbnail
       var getRegularThumbnail = async(function*() {
-        var sw = (yield navigator.serviceWorker.ready).active;
         var host = new URL(url).host;
         var thumbURL = new URL(`/pagethumbs/${host}`, window.location).href;
-        var hasSiteThumb = swMessage(sw, "NewTab:HasSiteThumb");
-        var thumbInCache = yield hasSiteThumb({thumbURL});
+        var thumbInCache = yield CacheTasks.has(thumbURL, "pagethumbs_cache");
```
Comment (oyiptong) [inline src/js/sites.js:204]: :+1:

**Comment by oyiptong** [comment]
big change. hard to review.
Could've been broken up in 2 or 3 PR's.

That said, I took a look and things look good.

r+

**Pull request merged**